modules/hal_rpi_pico/CMakeLists.txt

```diff
@@ -151,6 +151,18 @@ if(CONFIG_HAS_RPI_PICO)
   zephyr_include_directories_ifdef(CONFIG_PICOSDK_USE_CLAIM
           ${common_dir}/hardware_claim/include)
 
+  zephyr_library_sources_ifdef(CONFIG_PICOSDK_USE_RAND
+    ${rp2_common_dir}/hardware_timer/timer.c
```

Comment:
This needs thinking about, in particular, how this interacts, by default, with Pico SDK's alarm pool. I think `PICO_TIME_DEFAULT_ALARM_POOL_DISABLED` will need to be set, at least.